liquibase-standard/src/main/java/liquibase/change/custom/CustomChangeWrapper.java

```diff
@@ -192,14 +192,17 @@ public SqlStatement[] generateStatements(Database database) {
     @Override
     public SqlStatement[] generateRollbackStatements(Database database) throws RollbackImpossibleException {
         SqlStatement[] statements = null;
+        boolean shouldExecute = Scope.getCurrentScope().get(Change.SHOULD_EXECUTE, Boolean.TRUE);
         try {
             configureCustomChange();
-            if (customChange instanceof CustomSqlRollback) {
-                statements = ((CustomSqlRollback) customChange).generateRollbackStatements(database);
-            } else if (customChange instanceof CustomTaskRollback) {
-                ((CustomTaskRollback) customChange).rollback(database);
-            } else {
-                throw new RollbackImpossibleException("Unknown rollback type: " + customChange.getClass().getName());
+            if (shouldExecute) {
+                if (customChange instanceof CustomSqlRollback) {
+                    statements = ((CustomSqlRollback) customChange).generateRollbackStatements(database);
+                } else if (customChange instanceof CustomTaskRollback) {
+                    ((CustomTaskRollback) customChange).rollback(database);
+                } else {
+                    throw new RollbackImpossibleException("Unknown rollback type: " + customChange.getClass().getName());
+                }
             }
```

Comment:
same to follow code
https://github.com/liquibase/liquibase/blob/v4.29.1/liquibase-standard/src/main/java/liquibase/change/custom/CustomChangeWrapper.java#L163-L174